pkg/controller/subscription/subscription_controller.go

```diff
@@ -253,24 +245,36 @@ func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscrip
 
 	isServiceAffecting := rhmiConfigs.IsUpgradeServiceAffecting(latestRHMICSV)
 
-	// checks if there is changes in the rhmiconfig cr and recalculate the status
-	if reschedule, ok := config.Annotations[v1alpha1.RecalculateScheduleAnnotation]; ok && reschedule == "true" {
-		err = rhmiConfigs.UpdateStatus(ctx, r.client, config, latestRHMIInstallPlan, rhmiSubscription.Status.CurrentCSV)
-		if err != nil {
-			return reconcile.Result{}, err
+	if isServiceAffecting {
+
+		newUpgradeAvailable := &integreatlyv1alpha1.UpgradeAvailable{
+			TargetVersion: rhmiSubscription.Status.CurrentCSV,
+			AvailableAt:   latestRHMIInstallPlan.CreationTimestamp,
 		}
 
-		config.Annotations[v1alpha1.RecalculateScheduleAnnotation] = ""
-		err = r.client.Update(ctx, config)
-		if err != nil {
+		if !reflect.DeepEqual(newUpgradeAvailable, config.Status.UpgradeAvailable) {
```

Comment:
```suggestion
		if !reflect.DeepEqual(newUpgradeAvailable, config.Status.UpgradeAvailable) && !latestRHMIInstallPlan.Spec.Approved {
```

We have to add this condition, otherwise, once the upgrade is approved and the `UpgradeAvailable` field is set to empty, it will be set again in the next reconcilliation

Reply:
:+1: